packages/docs/services/inversify-site/versioned_docs/version-6.x/fundamentals/di-hierarchy.mdx

```diff
@@ -13,9 +15,17 @@ With InversifyJS's hierarchical injection system, you can create a hierarchy of
 
 When a dependency needs to be injected, InversifyJS starts by looking in the current container for a binding. If the binding is not found, it moves up the hierarchy to the parent container and continues the search. This process continues until a binding is found or the top-level parent container is reached.
 
-:::warning
+:::warning[Binding overrides]
+
+Found bindings might override ancestor bindings even if their constraints are not met. For example, if a named binding is found in the child container for the requested service, that binding overrides parent bindings even if this binding is later discarded in a non-named resolution request.
+
+:::
+
+:::warning[DI hierarchies and cached bindings]
+
+When using hierarchical injection, be aware that cached bindings keep cached values from the first resolution, even if the call comes from a child container.
```

Comment:
When using hierarchical injection, be aware that cached bindings ~~keep cached values~~ from the first resolution ***will be used for subsequent resolution***, even if the call comes from ~~a~~ ***another*** child container. 

***If this behaviour is unwanted, use `ContainerModule` instead.***

Reply:
`ContainerModules` are out of the scope of DI hierarchy docs, I will think about it